### PR TITLE
[kube-prometheus-stack] add alertmanager-instance-selector and thanos…

### PR DESCRIPTION
**Pull request by z-anshun**
…-ruler-instance-selector

#### What this PR does / why we need it
Currently chart only supports `select-field-selector`, but `prom-operator` also supports `alertmanager-instance-selector` and `thanos-ruler-instance-selector` options, and in some cases they can be set to reduce CPU and Memory usage .

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/values.yaml [hunk 1]
```diff
@@ -2215,6 +2215,10 @@ prometheusOperator:
     tag: v0.30.2
     sha: ""
 
+  ## Set a Label Selector to filter watched alertManager and thanosRuler
+  alertManagerSelector: ""
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/values.yaml:2219]: We usually prefer alertmanager over alertManager

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/values.yaml:2219]: Also this should be named alertmanagerInstanceSelector

File: charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml [hunk 2]
```diff
@@ -107,6 +107,12 @@ spec:
             {{- if .Values.prometheusOperator.thanosRulerInstanceNamespaces }}
             - --thanos-ruler-instance-namespaces={{ .Values.prometheusOperator.thanosRulerInstanceNamespaces | join "," }}
             {{- end }}
+            {{- if .Values.prometheusOperator.alertManagerSelector }}
+            - --alertmanager-instance-selector={{ .Values.prometheusOperator.alertManagerSelector }}
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml:111]: Can you move this with next to the --alertmanager-instance-namespaces flag?

File: charts/kube-prometheus-stack/Chart.yaml [hunk 3]
```diff
@@ -21,7 +21,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 45.7.1
+version: 45.8.0
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/Chart.yaml:24]: This needs a bump

File: charts/kube-prometheus-stack/values.yaml [hunk 4]
```diff
@@ -2215,6 +2215,10 @@ prometheusOperator:
...
+  thanosRulerSelector: ""
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/values.yaml:2220]: Can you also add this one prometheus-instance-selector? https://github.com/prometheus-operator/prometheus-operator/blob/27943fe43862d313a63b518b6ee67b59d7680c1d/cmd/operator/main.go#L175

Comment (QuentinBisson) [inline charts/kube-prometheus-stack/values.yaml:2220]: The thanosRulerSelector needs a comment

**Review by QuentinBisson — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/values.yaml (hunk 4 above)
Comment (QuentinBisson) [reply charts/kube-prometheus-stack/values.yaml:2234]: Done

File: charts/kube-prometheus-stack/values.yaml [hunk 5]
```diff
@@ -2222,6 +2222,17 @@ prometheusOperator:
     tag: v0.30.2
     sha: ""
 
+  ## Set a Label Selector to filter watched prometheus and prometheusAgent
+  ##
+  promSelector: ""
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/values.yaml:2227]: Can you name this prometheusInstanceSelector and add Instance to the other 2?

Comment (z-anshun) [reply charts/kube-prometheus-stack/values.yaml]: > Can you name this prometheusInstanceSelector and add Instance to the other 2?

Done

File: charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml [hunk 6]
```diff
@@ -93,12 +93,18 @@ spec:
             {{- if .Values.prometheusOperator.alertmanagerInstanceNamespaces }}
             - --alertmanager-instance-namespaces={{ .Values.prometheusOperator.alertmanagerInstanceNamespaces | join "," }}
             {{- end }}
+            {{- if .Values.prometheusOperator.alertmanagerSelector }}
+            - --alertmanager-instance-selector={{ .Values.prometheusOperator.alertmanagerSelector }}
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml:97]:
```suggestion
            - --alertmanager-instance-selector={{ .Values.prometheusOperator.alertmanagerInstanceSelector }}
```

File: charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml [hunk 7]
```diff
@@ -107,6 +113,9 @@ spec:
             {{- if .Values.prometheusOperator.thanosRulerInstanceNamespaces }}
             - --thanos-ruler-instance-namespaces={{ .Values.prometheusOperator.thanosRulerInstanceNamespaces | join "," }}
             {{- end }}
+            {{- if .Values.prometheusOperator.thanosRulerSelector }}
+            - --thanos-ruler-instance-selector={{ .Values.prometheusOperator.thanosRulerSelector }}
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml:117]:
```suggestion
            - --thanos-ruler-instance-selector={{ .Values.prometheusOperator.thanosRulerInstanceSelector }}
```

File: charts/kube-prometheus-stack/values.yaml [hunk 8]
```diff
@@ -2222,6 +2222,17 @@ prometheusOperator:
...
+  ## Set a Label Selector to filter watched prometheus and prometheusAgent
+  ##
+  prometheusInstanceSelector: ""
+
+  ## Set a Label Selector to filter watched alertmanager
+  ##
+  alertmanagerSelector: ""
+
+  ## Set a Label Selector to filter watched thanosRuler
+  thanosRulerSelector: ""
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/values.yaml:2234]:
```suggestion
  thanosRulerInstanceSelector: ""
```

File: charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml [hunk 9]
```diff
@@ -107,6 +113,9 @@ spec:
             {{- if .Values.prometheusOperator.thanosRulerInstanceNamespaces }}
             - --thanos-ruler-instance-namespaces={{ .Values.prometheusOperator.thanosRulerInstanceNamespaces | join "," }}
             {{- end }}
+            {{- if .Values.prometheusOperator.thanosRulerSelector }}
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml:116]:
```suggestion
            {{- if .Values.prometheusOperator.thanosRulerInstanceSelector }}
```

**Review by QuentinBisson — APPROVED** [review]

**Pull request merged**